core/common/src/geometry/ElementGeometry.ts

```diff
@@ -623,6 +637,251 @@ export namespace ElementGeometry {
+      case ElementGeometryOpcode.ArcPrimitive: {
+        const buffer = new flatbuffers.ByteBuffer(entry.data);
+        const ppfb = EGFBAccessors.ArcPrimitive.getRootAsArcPrimitive(buffer);
+        return (EGFBAccessors.BoundaryType.Closed === ppfb.boundary());
+      }
+
+      case ElementGeometryOpcode.CurvePrimitive: {
+        // should never be a closed bcurve...
+        return false;
+      }
+
+      case ElementGeometryOpcode.CurveCollection: {
+        const geom = BentleyGeometryFlatBuffer.bytesToGeometry(entry.data, true);
+        if (undefined === geom || Array.isArray(geom))
+          return false;
+        return ("curveCollection" === geom.geometryCategory && (geom as CurveCollection).isAnyRegionType);
+      }
+
+      case ElementGeometryOpcode.SolidPrimitive: {
+        const geom = BentleyGeometryFlatBuffer.bytesToGeometry(entry.data, true);
+        if (undefined === geom || Array.isArray(geom))
+          return false;
+        return ("solid" === geom.geometryCategory && !(geom as SolidPrimitive).isClosedVolume);
+      }
+
+      case ElementGeometryOpcode.Polyface: {
+        const geom = BentleyGeometryFlatBuffer.bytesToGeometry(entry.data, true);
+        if (undefined === geom || Array.isArray(geom))
+          return false;
+        if ("polyface" !== geom.geometryCategory)
+          return false;
+        const polyface = (geom as Polyface);
+        switch (polyface.expectedClosure) {
+          case 0:
+            return !PolyfaceQuery.isPolyfaceClosedByEdgePairing(polyface);
+          case 1:
+            return true;
+          case 2:
+          default:
+            return false;
+        }
+      }
+
+      case ElementGeometryOpcode.BsplineSurface: {
+        // never treated as a solid even if closed/periodic in u/v...
+        return true;
+      }
+
+      case ElementGeometryOpcode.BRep: {
+        const buffer = new flatbuffers.ByteBuffer(entry.data);
+        const ppfb = EGFBAccessors.BRepData.getRootAsBRepData(buffer);
+        return (EGFBAccessors.BRepType.Sheet === ppfb.brepType());
+      }
+
+      default:
+        return false;
+    }
+  }
+
+  /** Return whether the supplied entry represents a capped solid, closed polyface, or solid body */
+  export function isSolid(entry: ElementGeometryDataEntry): boolean {
+    switch (entry.opcode) {
+      case ElementGeometryOpcode.SolidPrimitive: {
+        const geom = BentleyGeometryFlatBuffer.bytesToGeometry(entry.data, true);
+        if (undefined === geom || Array.isArray(geom))
+          return false;
+        return ("solid" === geom.geometryCategory && (geom as SolidPrimitive).isClosedVolume);
+      }
+
+      case ElementGeometryOpcode.Polyface: {
+        const geom = BentleyGeometryFlatBuffer.bytesToGeometry(entry.data, true);
+        if (undefined === geom || Array.isArray(geom))
+          return false;
+        if ("polyface" !== geom.geometryCategory)
+          return false;
+        const polyface = (geom as Polyface);

```

Comment:
Just curious, this cast to Polyface isn't actually necessary because of check above it, right?

Reply:
It's necessary...can't do geom.expectedClosure...